### ISSUE TITLE
Support for in-place operations?

### ISSUE THREAD

**Issue by ntezak** [ahwillia/Einsum.jl#5]
Hi, would it fit to the overall vision for the package to support in-place operations for pre-allocated arrays?

``` julia
E.g., 
A = randn(5,6,7);

X = randn(5,2);
Y = randn(6,2);
Z = randn(7,2);

@einsum A[i,j,k] += X[i,r]*Y[j,r]*Z[k,r]
```

The code changes are minimal (I have already added them as well as testing code in my fork).
Here is the commit:
ntezak/Einsum.jl@73d087c36d7a1b2917bb82de0090bac698effa40

**Comment by ahwillia** [comment]
Absolutely! As long as it doesn't break anything.

Go ahead an open a PR with some tests.

**Comment by ahwillia** [comment]
Closed by #7
